### PR TITLE
chore: merge master in 6.2.x for 0.17 release

### PR DESCRIPTION
**Pull request by lct45**
### Description 
_What behavior do you want to change, why, how does your patch achieve the changes?_

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by vcrfxia — COMMENTED** [review]
@lct45 we can open this PR for folks to review the changes and to ensure the build passes, but we should not merge this PR unless we want to temporarily allow merge commits rather than squash commits. Squashing the commits will break our auto changelog generation.

**Review by JimGalasyn — APPROVED** [review]

**Review by stevenpyzhang — COMMENTED** [review]
We need to revert https://github.com/confluentinc/ksql/pull/7207/commits/dff3f0e650fb18c051b9c3ec41a96097e3effda1 as we don't want to change the project version in 6.2.x

At a glance, everything else looks alright

**Pull request merged**
